securedrop/tests/functional/app_navigators.py

```diff
@@ -0,0 +1,387 @@
+from contextlib import contextmanager
```

Comment:
Most of this is existing code from the `FunctionalTest` class that I've repackaged into separate classes to improve readability.